scipy/stats/_morestats.py

```diff
@@ -4333,16 +4333,21 @@ def median_test(*samples, ties='below', correction=True, lambda_=1,
     return MedianTestResult(stat, p, grand_median, table)
 
 
-def _circfuncs_common(samples, high, low):
+def _circfuncs_common(samples, high, low, xp=None):
+    xp = array_namespace(samples) if xp is None else xp
     # Ensure samples are array-like and size is not zero
-    if samples.size == 0:
-        NaN = _get_nan(samples)
+    if xp_size(samples) == 0:
+        NaN = _get_nan(samples, xp=xp)
         return NaN, NaN, NaN
 
+    if xp.isdtype(samples.dtype, 'integral'):
+        dtype = xp.asarray(1.).dtype  # get default float type
+        samples = xp.asarray(samples, dtype=dtype)
```

Comment:
I added this for one tests with `uint8` input. For most array types, the operations in the function turn these into floats naturally, but array_api_strict raises an error about not promoting uint8 blah blah. If we have to manually promote integers, does it make sense to promote them to the native float dtype (instead of always `float64`)? Should this be a helper function in `_array_api.py`?

Reply:
This seems like it would be helpful

Reply:
I'll let that be a separate PR.